_posts/not-all-changes-are-born-equal.md

```diff
@@ -0,0 +1,71 @@
+
+A real-life sample would be something like this:
+
+> If a company stores its manufactures into a warehouse, whether the warehouse is automated or not is irrelevant for other departments, e.g., marketing.
+
+In other words, any change to the internal implementation of the warehouse is not visible from the outside. They are decoupled.
+Another sample is a new discount or pricing policy. If to change the pricing policy, there is a need to change multiple services, e.g., a products service and a sales service because product prices are in products, and the discount logic is in sales, it's clear that we have a coupling issue.
+
+At this point, usually after a talk at a conference, people come to me with objections like this:
+
+> What if a company that manufactures cars decides to start manufacturing also clothes? That has an impact both on the marketing department and the warehouse.
+
+And many other services, I'd say. Usually, they continue with:
+
+> the presented design/solution/idea doesn't address this problem. See, they're coupled anyway, so why bothering.
+
+We're not here to solve that kind of problem. That's a disruptive business change that is not that different from creating a new company that requires everything fresh from scratch, including a software system. There is no point in designing a software architecture that tries to absorb this type of change nicely. It's not a coupling problem.
+
+As a side note, you can see the problem at play on many webshops. If you look for products defined by multiple attributes, like clothes, you'll notice how bad they are at managing them. The UI and the overall buying experience are not optimized for those types of products; The same goes for the search experience that doesn't factor in those attributes. Many sellers create a separate product for each attributes combination, which is a bit better for them. However, it is still a problem for customers because the same shoes end up being multiple product SKUs based on colors and sizes, and searching is always painful. It seems they prefer forcing sellers to adhere to the current status quo rather than coming up with a change that would probably affect many various services. That's understandable; the original domain was a different one, and the requested changes are disruptive because of the drastically different domain.
```

Comment:
```suggestion
As a side note, you can see the problem at play in many webshops. If you look for products defined by multiple attributes, like clothes, you'll notice how bad they are at managing them. The UI and the overall buying experience are not optimized for those types of products. The same goes for the search experience that doesn't factor in those attributes. Many sellers create a separate product for each combination of attributes, which is a bit better for them. However, it is still a problem for customers because the same shoes end up having multiple product SKUs based on different colors and sizes, and searching is always painful. It seems they prefer forcing sellers to adhere to the current status quo rather than coming up with a change that would probably affect many various services. That's understandable; the original domain was a different one, and the requested changes are disruptive because of the drastically different domain.
```